src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -17903,5 +17908,690 @@ public class XAttribute : System.Attribute
                 Diagnostic(ErrorCode.ERR_BadAttributeParamType, "X").WithArguments("a", "A").WithLocation(12, 2)
                 );
         }
+
+        [Fact]
+        public void Nullability_NullableElement_ArrayCollection()
+        {
+            string src = """
+                #nullable enable
+                string[] x1 = [null];
+                string?[] x2 = [null];
```

Comment:
Nullable analysis covers both nullable reference types and nullable value types, but for nullable value types it's mostly about tracking the state (so that when you do `.Value.Something()` we can warn you about unsafe dereferencing).
In collections scenarios, I don't see much interesting NVT scenarios. We'll mostly get binding errors.
I'll add a test to illustrate.

Reply:
well, it was more about validating that the following works:

```c#
ImmutableArray<byte?>? x = [0];
```

Reply:
Added